src/lib.rs

```diff
@@ -1649,20 +1660,42 @@ impl fmt::Display for RcgenError {
 			#[cfg(feature = "x509-parser")]
 			UnsupportedExtension => write!(f, "Unsupported extension requested in CSR")?,
 			RingUnspecified => write!(f, "Unspecified ring error")?,
-			RingKeyRejected(e) => write!(f, "Key rejected by ring: {}", e)?,
+			RingKeyRejected(reason) => write!(f, "Key rejected by ring: {}", reason)?,
 			CertificateKeyPairMismatch => write!(f, "The provided certificate's signature \
 				algorithm is incompatible with the given key pair")?,
 
 			Time => write!(f, "Time error")?,
 			RemoteKeyError => write!(f, "Remote key error")?,
 			#[cfg(feature = "pem")]
-			PemError(e) => write!(f, "PEM error: {}", e)?,
+			PemError(_) => write!(f, "PEM error")?,
```

Comment:
error doesn't contain any useful info?

Reply:
It does! But generally, it is more useful if an error returns its source via `source`. If you use something like `anyhow`, that will concatenate the various error messages for you. If you also include them in the `Display` implementation, you end up duplicating the message.

See https://github.com/rust-lang/project-error-handling/issues/23.

Reply:
Happy to drop that commit in case people disagree

Reply:
I share the concerns written by `yaahc` in that thread, I think we should not implement the `source` function.

> Thing's like logging or printing errors directly, or panicking with `Result`'s containing `E: Error` types that haven't been converted to a `Reporter` will fail to introduce the party that is responsible for iterating over the sources and printing all of the errors.